include/random_walks/hamiltonian_monte_carlo_walk.hpp

```diff
@@ -0,0 +1,168 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+// Copyright (c) 2020-2020 Marios Papachristou
+
+// Contributed and/or modified by Marios Papachristou, as part of Google Summer of Code 2020 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef HAMILTONIAN_MONTE_CARLO_WALK_HPP
+#define HAMILTONIAN_MONTE_CARLO_WALK_HPP
+
+
+#include "generators/boost_random_number_generator.hpp"
+#include "random_walks/gaussian_helpers.hpp"
+#include "ode_solvers/ode_solvers.hpp"
+
+struct HamiltonianMonteCarloWalk {
+
+  template
+  <
+    typename NT,
+    typename OracleFunctor
+  >
+  struct parameters {
+    NT epsilon; // tolerance in mixing
+    NT eta; // step size
+
+    parameters(
+      OracleFunctor const& F,
+      unsigned int dim,
+      NT epsilon_=1e-4)
+    {
+      epsilon = epsilon_;
+      eta = 1.0 / (sqrt(pow(dim, 3) * F.params.L));
```

Comment:
it would be useful to link the publication that provides an evidence for using this bound